Feat/camera check

marijnvdwerf opened the pull request:
Still in progress, but thought I’d share what I was working on.

marijnvdwerf commented:
@ethteck I think this should be 'good enough' to merge. Afterwards we can look at creating a `camera-check.h` and move some structs to that as a common header.

ethteck reviewed (changes requested):
After the merge conflict is cleared up, I'm happy to merge this. but I'd prefer we avoid any more foo/bar/fizz struct names and just go with unk8000whatever at least, if not the silly names. there's too many foo/bar/fizz and idk how any of them are different from each other

marijnvdwerf commented:
- Rebased
- Grouped files
- Added shared (private) header
- Namespaced names
- Added random names (If we're doing Unk80blabla, we might as well do Kanji. Both are equally unreadable to me).
